web/src/user_card_popover.ts

```diff
@@ -825,14 +849,28 @@ function register_click_handlers(): void {
     $(".buddy-list-section").on("click", ".user-list-sidebar-menu-icon", (e) => {
         e.stopPropagation();
         const $target = $(e.currentTarget).closest("li");
-
+        const user_id = elem_to_user_id($target);
+        const last_active = presence.last_active_date(user_id);
+        if (!last_active) {
+            presence.fetch_user_last_seen_status(user_id, () => {
+                const $target = $(e.currentTarget).closest("li");
+                toggle_sidebar_user_card_popover($target);
+            });
+        }
         toggle_sidebar_user_card_popover($target);
     });
 
     $(".buddy-list-section").on("click", ".user-profile-picture", (e) => {
         e.stopPropagation();
         const $target = $(e.currentTarget).closest("li");
-
+        const user_id = elem_to_user_id($target);
+        const last_active = presence.last_active_date(user_id);
+        if (!last_active) {
+            presence.fetch_user_last_seen_status(user_id, () => {
+                const $target = $(e.currentTarget).closest("li");
+                toggle_sidebar_user_card_popover($target);
```

Comment:
I think Tim mentioned to just open the popover and show "Loading..." indicator initially and then set the status after the data is fetched. Why is this function to toggle the popover is being pased as callback?

Reply:
And I think the function for fetching the presence from server can be called in `toggle_sidebar_user_card_popover`.

Reply:
I was displaying "Loading..." when the card opened initially. I am passing the `user_id` to the `fetch_user_last_seen_status` function, which fetches the user presence data and then calls the `toggle_sidebar_user_card_popover` function again, which I am passing as a callback.

As a result, the function for opening the user card is being called twice when the presence data is not readily available. I removed the `current_user_sidebar_user_id`-related code because it was causing the popover to hide when the `toggle_sidebar_user_card_popover` function was called a second time after the data fetch.

@**sahil839**, please let me know if there’s a better approach to improve the data fetching and callback mechanism.